src/js/cart.js

```diff
@@ -0,0 +1,143 @@
+const cartList = document.querySelector('.cart__table-content');
+const cartItems = document.querySelector('.cart-items');
+const cartTotal = document.querySelector('.cart-total');
+const clearCartBtn = document.querySelector('.cart-clear');
+const updateCartBtn = document.querySelector('.cart-update');
+let cart = [];
+let cartItemEvent = [];
+
+class View {
+  initialApp() {
+    cart = Storage.getCart();
+    this.setCartValues();
+  }
+
+  setCartValues() {
+    let tempTotal = 0;
+    let itemsTotal = 0;
+    cart.map((item) => {
+      tempTotal += item.price * item.amount;
+      itemsTotal += item.amount;
+    });
+    cartItems.innerText = `Tổng số tiền lượng sản phẩm : ${itemsTotal}`;
+    cartTotal.innerText = `Tổng số tiền  : ${tempTotal}`;
+  }
+
+  removeItem(id) {
+    cart = cart.filter((item) => item.id !== id);
+    this.setCartValues();
+    Storage.saveCart(cart);
+  }
+
+  clearCart() {
+    cart = [];
+    this.setCartValues();
+    Storage.saveCart(cart);
+    cartList.innerHTML = '';
+  }
+
+  renderCart() {
+    cart.map((cartItem) => {
+      const cartHTML = `
+          <tr class="table__tr cart-item" data-id=${cartItem.id}> 
+            <td class="table__td" data-title="Ảnh"><img class="table__img" src="${
+              cartItem.src
+            }" alt=""></td>
+            <td class="table__td" data-title="Tên Sản Phẩm">${
+              cartItem.title
+            }</td>
+            <td class="table__td table__price" data-title="Giá">${
+              cartItem.price
+            } VND</td>
+            <td class="table__td" data-title="Số lượng">
+              <i class="fa fa-plus cart-up" data-id=${cartItem.id}></i>  
+              <span class="px-3  table__amount" data-id=${cartItem.id} >${
+        cartItem.amount
+      }</span>
+              <i class="fa fa-minus cart-down" data-id=${cartItem.id}></i> 
+            </td>
+            <td class="table__td table__total" data-title="Tổng số">${
+              cartItem.price * cartItem.amount
+            } VNĐ</td>
+            <td class="table__td" data-title="Xóa" data-id=${cartItem.id}> 
+                <button class="delete-product btn button" data-id=${
+                  cartItem.id
+                }><i class="fa fa-trash" ></i></button>
+            </td>
+        </tr>
+          `;
+      cartList.insertAdjacentHTML('beforeend', cartHTML);
+    });
+  }
+
+  updateCart(productId, amount, total) {
+    const itemAmountList = [...document.querySelectorAll('.table__amount')];
+    const itemSumList = [...document.querySelectorAll('.table__total')];
+    itemAmountList.forEach((item) => {
```

Comment:
for với map khác nhau chỗ nào

Reply:
forEach thì trả về mảng ban đầu có thể thay đổi trực tiếp, còn map thì trả về mảng mới có giá trị như mảng cũ đã thay đổi